packages/moonstone/internal/DateTimeDecorator/DateTimeDecorator.js

```diff
@@ -105,10 +109,14 @@ const DateTimeDecorator = hoc((config, Wrapped) => {
 
 		static getDerivedStateFromProps (props, state) {
 			const propValue = toTime(props.value);
-			if (state.value !== propValue) {
+			if (state.derived && state.value !== propValue) {
```

Comment:
Maybe it would be better to use a `typeof props.value !== 'undefined'` to avoid applying bad values?